server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java

```diff
@@ -405,6 +461,172 @@ public void vaultEndpointShouldRestoreMessageForSharer() {
             .body(ARGUMENTS + ".list.subject", hasItem(SUBJECT));
     }
 
+    @Category(BasicFeature.class)
+    @Test
+    public void vaultExportShouldSendMailsToShareeWhenJmapDeleteMessage() throws Exception {
+        bartSendMessageToHomer();
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 1);
+        String messageIdOfHomer = listMessageIdsForAccount(homerAccessToken).get(0);
+
+        homerDeletesMessages(listMessageIdsForAccount(homerAccessToken));
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 0);
+
+        String fileLocation = exportAndGetFileLocationFromLastMail(EXPORT_ALL_HOMER_MESSAGES_TO_BART, bartAccessToken);
+
+        try (ZipFile zipFile = zipFile(fileSystem.getResource(fileLocation))) {
```

Comment:
```
try (ZipAssert zipAssert = assertThatZip(fileSystem.getResource(fileLocation)) {
    zipAssert.containsOnlyEntriesMatching(hasName(messageIdOfHomer));
}
```

We may need to write an `assertThatZip` method taking an inputstream and doing an inmemory copy...